tests/lib/rules/valid-jsdoc.js

```diff
@@ -287,6 +287,22 @@ eslintTester.addRuleTest("lib/rules/valid-jsdoc", {
                 message: "Unexpected @returns tag; function has no return statement.",
                 type: "Block"
             }]
+        },
+        {
+            code: "/**\n * Some text\n *\n * @param arg [optional] text description\n */\nfunction testFunction(arg) {}",
```

Comment:
Then it seems you've solved a different (unreported) bug, but the one in #2270 still is not resolved. So you can either open a new issue describing this bug, or add the description to #2270 and include a fix for that bug in this PR.